src/Servers/Kestrel/Transport.MsQuic/src/Internal/MsQuicNativeMethods.cs

```diff
@@ -470,6 +471,114 @@ internal unsafe struct QuicBuffer
             internal byte* Buffer;
         }
 
+        public static SOCKADDR_INET Convert(IPEndPoint endpoint)
+        {
+            var lResult = ConvertToSocketAddrInet(endpoint.Address);
+            SetPort(endpoint.Address.AddressFamily, ref lResult, endpoint.Port);
+            return lResult;
+        }
+
+        private static void SetPort(AddressFamily addressFamily, ref SOCKADDR_INET socketAddrInet, int originalPort)
+        {
+            if (0 > originalPort || originalPort > ushort.MaxValue)
+            {
+                throw new ArgumentOutOfRangeException(nameof(originalPort), originalPort, "invalid port");
+            }
+            var convertedPort = (ushort)IPAddress.HostToNetworkOrder((short)originalPort);
+            switch (addressFamily)
+            {
+                case AddressFamily.InterNetwork:
+                    socketAddrInet.Ipv4.sin_port = convertedPort;
+                    break;
+                case AddressFamily.InterNetworkV6:
+                    socketAddrInet.Ipv6.sin6_port = convertedPort;
+                    break;
+                default:
+                    socketAddrInet.Ipv6.sin6_port = convertedPort;
+                    break;
+            }
+        }
+
+        private const ushort AF_INET = 2;
+        private const ushort AF_INET6 = 23;
+        private const ushort AF_UNSPEC = 0;
+
+        private static SOCKADDR_INET ConvertToSocketAddrInet(IPAddress ipAddress)
+        {
+            var lResult = new SOCKADDR_INET();
+            switch (ipAddress.AddressFamily)
+            {
+                case AddressFamily.InterNetwork:
+                    lResult.Ipv4.sin_addr0 = ipAddress.GetAddressBytes()[0];
```

Comment:
Doesn't this allocate a crazy number of arrays?

Reply:
I need to play around with PhysicalAddress here.